cmis/src/test/java/nl/nn/adapterframework/extensions/cmis/TestGetAction.java

```diff
@@ -44,71 +38,51 @@ public class TestGetAction extends CmisSenderTestBase {
-				{ "atompub", "get", INPUT_WITH_PROPERTIES, GET_RESULT_FOR_GET_PROPERTIES, false, true, true},
-
-				{ "webservices", "get", INPUT_WITH_PROPERTIES, GET_RESULT_FOR_INPUT, false, false, false},
-				{ "webservices", "get", INPUT_WITH_PROPERTIES, GET_RESULT_TO_SERVLET, true, false, false},
-				{ "webservices", "get", INPUT_WITH_PROPERTIES, GET_RESULT_FOR_GET_PROPERTIES, false, true, false},
-				{ "webservices", "get", INPUT_WITH_PROPERTIES, GET_RESULT_FOR_GET_PROPERTIES, false, true, true},
-
-				{ "browser", "get", INPUT_WITH_PROPERTIES, GET_RESULT_FOR_INPUT, false, false, false},
-				{ "browser", "get", INPUT_WITH_PROPERTIES, GET_RESULT_TO_SERVLET, true, false, false},
-				{ "browser", "get", INPUT_WITH_PROPERTIES, GET_RESULT_FOR_GET_PROPERTIES, false, true, false},
-				{ "browser", "get", INPUT_WITH_PROPERTIES, GET_RESULT_FOR_GET_PROPERTIES, false, true, true},
-		});
+	public static Stream<Arguments> allImplementations() {
+		return Stream.of(
+				Arguments.of(BindingTypes.ATOMPUB, GET_RESULT_FOR_INPUT, false, false, false),
+				Arguments.of(BindingTypes.ATOMPUB, GET_RESULT_TO_SERVLET, true, false, false),
+				Arguments.of(BindingTypes.ATOMPUB, GET_RESULT_FOR_GET_PROPERTIES, false, true, false),
+				Arguments.of(BindingTypes.ATOMPUB, GET_RESULT_FOR_GET_PROPERTIES, false, true, true),
+
+				Arguments.of(BindingTypes.WEBSERVICES, GET_RESULT_FOR_INPUT, false, false, false),
+				Arguments.of(BindingTypes.WEBSERVICES, GET_RESULT_TO_SERVLET, true, false, false),
+				Arguments.of(BindingTypes.WEBSERVICES, GET_RESULT_FOR_GET_PROPERTIES, false, true, false),
+				Arguments.of(BindingTypes.WEBSERVICES, GET_RESULT_FOR_GET_PROPERTIES, false, true, true),
+
+				Arguments.of(BindingTypes.BROWSER, GET_RESULT_FOR_INPUT, false, false, false),
+				Arguments.of(BindingTypes.BROWSER, GET_RESULT_TO_SERVLET, true, false, false),
+				Arguments.of(BindingTypes.BROWSER, GET_RESULT_FOR_GET_PROPERTIES, false, true, false),
+				Arguments.of(BindingTypes.BROWSER, GET_RESULT_FOR_GET_PROPERTIES, false, true, true)
+		);
 	}
 
-	public TestGetAction(String bindingType, String action, String input, String expected, Boolean resultToServlet, Boolean getProperties, Boolean getDocumentContent) {
-		this.bindingType = bindingType;
-		this.action = action;
-		this.input = new Message(input);
-		this.expectedResult = expected;
-		this.resultToServlet = resultToServlet;
-		this.getProperties = getProperties;
-		this.getDocumentContent = getDocumentContent;
+	@ParameterizedTest(name = "{0} - {1} - toServlet = {2} - getProperties = {3} - getDocumentContent = {4}")
+	@MethodSource("allImplementations")
+	@Retention(RetentionPolicy.RUNTIME)
+	private @interface TestAllImplementations {
 	}
 
-	public void configure() throws Exception {
+	private void configure(BindingTypes bindingType, Boolean resultToServlet, Boolean getProperties, Boolean getDocumentContent) throws Exception {
```

Comment:
Ik zou het leesbaarder vinden als in al deze `configure()` methods ook de `sender` als parameter zou zijn meegegeven.

Reply:
Die zit 3 lagen diep...